ui/src/app/pages/data/cohort/cohort-criteria-menu.tsx

```diff
@@ -392,134 +392,139 @@ export const CohortCriteriaMenu = withCurrentWorkspace()(
                         ? optionList[0].category !== 'Program Data'
                         : optionList
                     )
-                    .map((category, index) => (
-                      <ul key={index}>
-                        {showMenuItem(index) && (
-                          <li
-                            style={styles.dropdownHeader}
-                            className='menuitem-header'
-                          >
-                            <span style={styles.dropdownHeaderText}>
-                              {category[0].category}
-                            </span>
-                          </li>
-                        )}
-                        {category
-                          .filter(
-                            (menuItem) =>
-                              domainCounts === null ||
-                              domainCounts.find(
-                                (dc) => dc.domain === menuItem.domain
-                              )
-                          )
-                          .map((menuItem, m) => (
+                    .map(
+                      (category, index) =>
+                        showMenuItem(index) && (
```

Comment:
Moved the `showMenuItem` check up a level to hide the entire `<ul>` for the category instead of just the first `<li>`